doc/crypto/overview/conventions.rst

```diff
@@ -370,25 +370,45 @@ In some environments, an application can make calls to the |API| in
 separate threads. In such an environment, *concurrent calls* are two or more
 calls to the API whose execution can overlap in time.
 
-Concurrent calls are performed correctly, as if the calls were executed in
-sequence, provided that they obey the following constraints:
-
-*   There is no overlap between an output parameter of one call and an input or
-    output parameter of another call. Overlap between input parameters is
-    permitted.
-
-*   A call to destroy a key must not overlap with a concurrent call to any of
-    the following functions:
-
-    *   Any call where the same key identifier is a parameter to the call.
-    *   Any call in a multi-part operation, where the same key identifier was
-        used as a parameter to a previous step in the multi-part operation.
-
-*   Concurrent calls must not use the same operation object.
-
-If any of these constraints are violated, the behavior is undefined.
-
-If the application modifies an input parameter while a function call is in
-progress, the behavior is undefined.
+**Sequential consistency**
+    The result of two or more concurrent calls must be consistent with the
+    same set of calls being executed sequentially in some order, provided that
+    the calls obey the following constraints:
+
+    *   There is no overlap between an output parameter of one call and an
+        input or output parameter of another call. Overlap between input
+        parameters is permitted.
+
+    *   A call to :code:`psa_destroy_key()` must not overlap with a concurrent
+        call to any of the following functions:
+
+        -   Any call where the same key identifier is a parameter to the call.
+        -   Any call in a multi-part operation, where the same key identifier
+            was used as a parameter to a previous step in the multi-part
+            operation.
+
+    *   Concurrent calls must not use the same operation object.
+
+    If any of these constraints are violated, the behavior is undefined.
+
+    The consistency requirement does not apply to errors that arise
+    from resource failures or limitations. For example, errors resulting from
+    resource constraints can arise in concurrent execution that do not arise in
```

Comment:
Minor: “constraints” is vague, I think “exhaustion” would make the example clearer.

Reply:
Fixed